### PR TITLE
docket: hash-based glob-ames, handled within app

### PR DESCRIPTION
**Pull request by Fang-**
This works around the mark definition issue that doing the same in spider would face, and makes versioning of globs explicit in the ames case as well, letting the docket file trigger new glob fetching.

Still missing:
- [x] Auto-updating the .docket file for a desk when a new glob is uploaded from the upload page.
- [ ] Treaty `+on-load` logic. Do we care about this or do we just nuke the testnet again?

Opening the PR now for review expedience.

### CODE REVIEW

**Review by belisarius222 — COMMENTED** [review]
Hm, intuitively, I would expect the hash to be there as a sanity check no matter whether it's downloaded over Ames or HTTP.  Would be it too much of a pain to require devs to put the hash in there if they're hosting globs over HTTP?

**Comment by Fang-** [comment]
We could probably do that. `-make-glob` just creates a file whose name is the hash, so shouldn't be hard on devs at all.

File: pkg/garden/ted/glob.hoon [hunk 1]
```diff
@@ -9,7 +9,7 @@
 |^
 ?-  -.loc
   %http  (fetch-http url.loc)
-  %ames  (fetch-ames ship.loc base)
+  %ames  !!  ::NOTE  done within docket itself
```
Comment (ryjm) [inline pkg/garden/ted/glob.hoon:12]: should we add a reminder to move this logic back here when the desk routing in gall is done?

**Comment by Fang-** [comment]
Alright I haven't tested the remote distribution path here very extensively yet, but at least locally this should behave a lot better now. ptal

File: pkg/arvo/lib/hood/kiln.hoon [hunk 2]
```diff
@@ -485,9 +485,9 @@
       ~>  %slog.0^leaf/"kiln: already tracking {here:(abed lac)}, ignoring"
       vats
     =:  loc  lac
-        rak  [~ *rein]
+        rak  [`[| our lac *aeon ~] *rein]
```
Comment (Fang-) [inline pkg/arvo/lib/hood/kiln.hoon:488]: This should probably be reverted in favor of 8d91e8a ?

File: pkg/garden/app/docket.hoon [hunk 3]
```diff
@@ -235,7 +271,35 @@
...
+        [%glob-ames ~]
+      ?-  -.sign
+        %kick      `state
+        %poke-ack  ~&([dap.bowl %unexpected-poke-ack] `state)
+      ::
+          %watch-ack
+        ?~  p.sign  `state
+        %-  %-  slog
+            leaf+"docket: failed to fetch glob over ames for {(trip desk)}"
+        ::TODO  maybe we want to retry, but with delay?
```
Comment (belisarius222) [inline pkg/garden/app/docket.hoon:283]: I think this should cancel the installation.  Docket should display a notification that the app failed to install, and stop trying to install it.  The user could ask around and try again or something, but the system shouldn't keep trying indefinitely if something is broken on the remote.

File: pkg/garden/ted/glob.hoon (hunk 1 above)
Comment (belisarius222) [reply pkg/garden/ted/glob.hoon:12]: Sure.  If we later fetch globs file by file, then doing that in a thread would probably be better.

**Review by belisarius222 — CHANGES_REQUESTED** [review]
We should:
- pull the glob hash out of the `$%`, since we'll want that there no matter how we fetch it
- write a migration, so we don't have to breach the testnet

**Pull request merged**